project/api/models.py

```diff
@@ -116,13 +116,16 @@ def clean(self, *args, **kwargs):
             raise ValidationError(
                 "Only Polygon and MultiPolygon objects are acceptable geometry types.")
 
-        # This date check is inculsive.
-        if Territory.objects.filter(
-                start_date__lte=self.end_date,
-                end_date__gte=self.start_date,
-                entity__exact=self.entity).exists():
-            raise ValidationError(
-                "Another territory of this PoliticalEntity exists during this timeframe.")
+        try:
+            # This date check is inculsive.
+            if Territory.objects.filter(
+                    start_date__lte=self.end_date,
+                    end_date__gte=self.start_date,
+                    entity__exact=self.entity).exists():
+                raise ValidationError(
+                    "Another territory of this PoliticalEntity exists during this timeframe.")
+        except Entity.DoesNotExist:
+            pass
```

Comment:
We still want to raise a validation error if they don't pass an entity.

`raise ValidationError("A territory must belong to an entity.")`

Reply:
Django does this automatically due to the non-null constraint.